rabbit-sink/src/main/java/org/springframework/cloud/stream/module/rabbit/sink/RabbitSink.java

```diff
@@ -0,0 +1,103 @@
+import org.springframework.amqp.support.converter.MessageConverter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.config.SpelExpressionConverterConfiguration;
+import org.springframework.cloud.stream.messaging.Sink;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.integration.amqp.outbound.AmqpOutboundEndpoint;
+import org.springframework.integration.amqp.support.DefaultAmqpHeaderMapper;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.util.StringUtils;
+
+/**
+ * A sink module that sends data to RabbitMQ.
+ *
+ * @author Gary Russell
+ */
+@EnableBinding(Sink.class)
+@EnableConfigurationProperties(RabbitSinkProperties.class)
+@Import(SpelExpressionConverterConfiguration.class)
+public class RabbitSink {
+
+	@Autowired
+	@Bindings(RabbitSink.class)
+	private Sink channels;
+
+	@Autowired
+	private RabbitSinkProperties properties;
+
+	@Autowired
+	private ApplicationContext context;
+
+	@ServiceActivator(inputChannel = Sink.INPUT)
+	@Bean
+	public MessageHandler amqpChannelAdapter(ConnectionFactory rabbitConnectionFactory) {
+		AmqpOutboundEndpoint handler = new AmqpOutboundEndpoint(rabbitTemplate(rabbitConnectionFactory));
+		DefaultAmqpHeaderMapper mapper = new DefaultAmqpHeaderMapper();
+		mapper.setRequestHeaderNames(this.properties.getMappedRequestHeaders());
+		handler.setHeaderMapper(mapper);
+		handler.setDefaultDeliveryMode(this.properties.getDeliveryMode());
+		if (this.properties.getExchangeExpression() == null) {
+			handler.setExchangeName(this.properties.getExchange());
+		}
+		else {
+			handler.setExpressionExchangeName(this.properties.getExchangeExpression());
+		}
+		if (this.properties.getRoutingKeyExpression() == null) {
+			handler.setRoutingKey(this.properties.getRoutingKey());
+		}
+		else {
+			handler.setExpressionRoutingKey(this.properties.getRoutingKeyExpression());
+		}
+		return handler;
+	}
+
+	@Bean
+	public RabbitTemplate rabbitTemplate(ConnectionFactory rabbitConnectionFactory) {
+		RabbitTemplate rabbitTemplate = new RabbitTemplate(rabbitConnectionFactory);
+		String converterBeanName = this.properties.getConverterBeanName();
+		if (StringUtils.hasText(converterBeanName)) {
+			if (RabbitSinkProperties.JSON_CONVERTER.equals(converterBeanName)) {
+				rabbitTemplate.setMessageConverter(jsonConverter());
+			}
+			else {
+				rabbitTemplate.setMessageConverter(this.context.getBean(converterBeanName, MessageConverter.class));
```

Comment:
I found that this looks better:
```
@Value("#{${converterBeanName:null}}")
private MessageConverter messageConverter;
```
Much closer to that what we can have in the XML config. :smile: